Assets/Scripts/UI/UIScrollers/HorizontalScroller.cs

```diff
@@ -0,0 +1,137 @@
+using DN.SceneManagement;
+
+namespace DN.UI
+{
+    ///<summary>
+    ///In this class i check wich button is the closest to the conter and snap it to that button.
+    ///I also set the sizes of the buttons and the interactivability here.
+    ///<summary>
+    public class HorizontalScroller : MonoBehaviour
+    {
+        public int CurrentBtnIndex => currentBtnIndex;
+
+        [SerializeField] private RectTransform panel;
+        [SerializeField] private RectTransform center;
+
+        [SerializeField] private GameObject parent;
+
+        [SerializeField] private LevelsData levelsData;
+        [SerializeField] private HorizontalScrollerLevelLoader horizontalScrollerLevelLoader;
+
+        private float[] distance;
+        private bool dragging = false;
+
+        private int btnDistance;
+        private int minBtnNumb;
+        private int currentBtnIndex = 0;
+
+        private bool checkedOnce = false;
+
+        private Vector2 minSize = new Vector2(150, 150);
+        private Vector2 maxSize = new Vector2(250, 250);
+
+        private const int ORIGIN_DISTANCE = 510;
+        private const float LERP_SPEED = 10f;
+
+        private List<RectTransform> currentRect = new List<RectTransform>();
+        private List<Button> currentButton = new List<Button>();
+        private List<Button> btn = new List<Button>();
+
+        private void Start()
+        {
+            int btnLength = btn.Count;
+            distance = new float[btnLength];
+
+            for (int i = 0; i < btn.Count; i++)
+            {
+                currentRect.Add(btn[i].transform.GetChild(0).GetComponent<RectTransform>());
+                currentButton.Add(btn[i].transform.GetChild(0).GetComponent<Button>());
+            }
+        }
+
+        private void Update()
+        {
+            if (!dragging)
+            {
+                LerpToBtn(minBtnNumb * -btnDistance);
+                if (checkedOnce)
+                {
+                    checkedOnce = true;
+                    return;
+                }
+            }
+
+            if (btnDistance != ORIGIN_DISTANCE)
+            {
+                CalculateDistance();
+            }
+
+            for (int i = 0; i < btn.Count; i++)
+            {
+                distance[i] = Mathf.Abs(center.transform.position.x - btn[i].transform.position.x);
+            }
+
+            float minDistance = Mathf.Min(distance);
+
+            for (int j = 0; j < btn.Count; j++)
+            {
+                if (minDistance == distance[j])
+                {
+                    minBtnNumb = j;
+                    currentBtnIndex = j;
+                    LerpBetweenButtonsSize(j, maxSize);
+                    SetButtonInteractive(j, !levelsData.Levels[j].IsLocked);
+                }
+                else
+                {
+                    LerpBetweenButtonsSize(j, minSize);
+                    SetButtonInteractive(j, !levelsData.Levels[j].IsLocked);
+                }
+            }
```

Comment:
Maak hier een aparte functie van, hij is tenslotte een beetje lang. Dan hoef je geen `if (checkedOnce)` te gebruiken en kan je deze gewoon callen in de `Awake()` en in de `Update()`. Bespaard een if check (ookal duurt dat maar 0.001 nanoseconde ofzo)